src/metabase/util/log.clj

```diff
@@ -48,16 +66,52 @@
          (lambdaisland.glogi/log logger# level# nil s#)
          a#))))
 
+(defn with-thread-context-fn
```

Comment:
What do you think of passing this an extra arg so its even more difficult to call outside the macro? Kind of like a lock and key.